Formula/o/octave.rb

```diff
@@ -152,14 +152,14 @@ def install
       { return ovl (42); }
     CPP
     system bin/"octave", "--eval", <<~MATLAB
-      mkoctfile ('-v', '-std=c++11', '-L#{lib}/octave/#{version}', 'oct_demo.cc');
+      mkoctfile ('-v', '-std=c++17', '-L#{lib}/octave/#{version}', 'oct_demo.cc');
```

Comment:
``` suggestion
      mkoctfile ('-v', '-L#{lib}/octave/#{version}', 'oct_demo.cc');
```

The build system should have figured out automatically which flags are needed to tell the compiler to use the C++17 standard. The necessary flags should be stored and used by `mkoctfile` automatically. No need to override it here.
If it turns out that you *do* need to override that here, please let me know. In any case, you should build with GNU extensions, i.e., `-std=gnu++17`. But like written previously, preferably don't override the C++ standard flags at all.

Reply:
When I remove this, I got an error and so it is needed now.

Reply:
> When I remove this, I got an error and so it is needed now.

That is unexpected.
Is it possible to see the output of the configure script when building Octave? At least the summary might already be helpful.

Does Homebrew set a `CXX` environment variable when building/testing the packages? If it does, which value does it have?

Reply:
If my suspicion about the failing dynare tests is correct, the following change might also fix this:
``` suggestion
      unsetenv ('CXX');
      mkoctfile ('-v', '-L#{lib}/octave/#{version}', 'oct_demo.cc');
```

Reply:
Unsetting `CXX` for the duration of the tests of Octave might still be a nice modification.
Users shouldn't need to pass `-std=gnu++17` every time they call `mkoctfile`. They also shouldn't need to set `CXX` manually.
It would be nice if the tests here could check if that works out of the box...
If `CXX` needs to be set as an environment variable for the recipe for some reason, it might be best if you could append ` -std=gnu++17` to `CXX` instead of passing the flag manually to `mkoctfile`.